packages/scripts/src/lib/_auto-nocheck.ts

```diff
@@ -0,0 +1,160 @@
+import { relative, dirname } from 'node:path';
+import { readFile, writeFile } from 'node:fs/promises';
+import { fileURLToPath } from 'node:url';
+import resolvePkg from 'resolve';
+import globPkg from 'glob';
+import yargs from 'yargs';
+import ora from 'ora';
+import { type ProjectAnalysis } from '@glint/core';
+
+const globSync = globPkg.sync;
+const resolve = resolvePkg.sync;
+
+export async function autoNocheck(
+  args: Array<string>,
+  { cwd = process.cwd(), spinner = ora() } = {}
+): Promise<void> {
+  let glint = await loadGlintCore(cwd);
+  let { globs, explanation } = parseArgs(args);
+
+  spinner.start('Starting Glint language service...');
+
+  let project = glint.analyzeProject(cwd);
+  let fileUpdates = new Map<string, string>();
+
+  for (let filePath of collectFilePaths(globs, cwd)) {
+    spinner.text = `Checking ${relative(cwd, filePath)}...`;
+
+    let fileContents = await readFile(filePath, 'utf-8');
+    let templatesWithErrors = findTemplatesWithErrors(glint, filePath, fileContents, project);
+    if (templatesWithErrors.size) {
+      fileUpdates.set(
+        filePath,
+        insertNocheckComments(filePath, fileContents, explanation, templatesWithErrors)
+      );
+    }
+  }
+
+  project.shutdown();
+
+  spinner.text = 'Writing `@glint-nocheck` comments...';
+
+  await writeFileContents(fileUpdates);
+
+  spinner.succeed(`Done! ${fileUpdates.size} files updated.`);
+}
+
+function parseArgs(args: Array<string>): { globs: Array<string>; explanation: string } {
+  return yargs(args)
+    .scriptName('auto-nocheck')
+    .command('$0 <globs...>', 'Write a description here', (command) =>
+      command
+        .positional('globs', {
+          type: 'string',
+          array: true,
+          describe: 'One or more paths or globs specifying the files to act on.',
+          demandOption: true,
+        })
+        .option('explanation', {
+          type: 'string',
+          default: 'not typesafe yet',
+          describe: 'The explanation to be included in @glint-nocheck comments',
+        })
+    )
+    .wrap(100)
+    .strict()
+    .parseSync();
+}
+
+type GlintCore = typeof import('@glint/core') &
+  typeof import('@glint/core/lib/language-server/util/index.js');
```

Comment:
Is this to avoid a direct dependency on it, or what? 🤔 Mind adding a comment so future-us remembers?

Reply:
Big-picture, this whole chunk is to make sure we're using the project-local version of `@glint/core`.

These two lines specifically are because I find `type FooMod = typeof import('foo')` a more intuitive way of capturing "the type of this entire module" than `import * as foo from 'foo'; type FooMod = typeof foo'`.

I'll add a note.

Reply:
Ah, that's fair and makes sense 👍🏼